authentication/oauth.go

```diff
@@ -0,0 +1,80 @@
+package authentication
+
+import (
+	"context"
+	"errors"
+	"net/url"
+
+	"github.com/auth0/go-auth0/authentication/oauth"
+)
+
+// OAuth exposes logging in using OAuth based APIs.
+type OAuth manager
+
+// LoginWithGrant allows logging in with an OAuth 2.0 grant. This should only be needed if a grant
+// type is not supported byt this SDK.
+func (o *OAuth) LoginWithGrant(ctx context.Context, grantType string, body url.Values, opts ...RequestOption) (t *oauth.TokenSet, err error) {
+	body.Add("grant_type", grantType)
+
+	err = o.authentication.FormRequest(ctx, "POST", o.authentication.URI("oauth", "token"), body, &t, opts...)
+	return
+}
+
+// LoginWithPassword is for logging in with information is typically received from a highly trusted
+// public client like a SPA.
```

Comment:
`a highly trusted public client like a SPA.` is this intended?

Reply:
I added this based on the docs in the [API docs](https://auth0.com/docs/api/authentication?http#resource-owner-password:~:text=typically%20using%20an%20interactive%20form%20in%20the%20user%2Dagent%20(browser)) and [node-auth0](https://github.com/auth0/node-auth0/blob/beta/src/auth/oauth.ts#L322)

Reply:
Just note that SPAs, being public clients, cannot hold a secret. So I'd suggest double-checking with @adamjmcgrath whether it's correct to call them "highly-trusted".

Reply:
Yep - that should read "This flow should only be used from highly-trusted applications that cannot do redirects" - no idea where I got that from in node-auth0 😅  will fix

Reply:
It's "typically received" which means the un/pw is _received_ from a public client - the grant itself is _sent_ from a confidential client

Although it does read a bit strange, let me take another look at it

Reply:
I think the API docs have changed since that node-auth0 docstring and they read a bit better - I would go with that 

```
This is the OAuth 2.0 grant that highly-trusted apps use to access an API. In this flow, the end-user is asked to fill in credentials (username/password), typically using an interactive form in the user-agent (browser). This information is sent to the backend and from there to Auth0. It is therefore imperative that the application is absolutely trusted with this information. For [single-page applications and native/mobile apps](https://auth0.com/docs/flows/concepts/auth-code-pkce), we recommend using web flows instead.
```

Reply:
"Highly Trusted public client like a SPA" in this instance means, a browser based client you trust well enough to handle usernames and passwords (like lock.js) - but the updated API docs is better so I'd go with that